hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/handle/record/GenesisRecordsConsensusHook.java

```diff
@@ -132,16 +133,18 @@ private void createAccountRecordBuilders(
             @NonNull final TokenContext context,
             @Nullable final String recordMemo,
             @Nullable final Long overrideAutoRenewPeriod) {
-        for (Map.Entry<Account, CryptoCreateTransactionBody.Builder> entry : map.entrySet()) {
+        final var orderedAccts = map.keySet().stream()
+                .sorted(Comparator.comparingLong(acct -> acct.accountId().accountNum()))
```

Comment:
This will only work as long as we have no shards or realms (it will be unspecified ordering for same-number accounts once we have multiple shards and/or realms).
OK for now, but making most of the protobuf ID classes comparable is probably a good idea long-term so we don't need adhoc comparators like this.

Reply:
Good call. I don't see any `Comparable` in the `AccountID` class, any idea where it is?

Reply:
Not there yet.  We need to work with Lubo to have a comment directive added to the PBJ compiler so it can generate the compareTo method and add the Comparable interface on selected classes; then add that directive to ID classes and a few others.
That is not going to happen tomorrow, so this approach will have to do for now.